youtube_dl/archive.py

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env python
+# coding: utf-8
+import errno
+import os
+
+from youtube_dl.compat import compat_st_mtime
+from youtube_dl.utils import locked_file
+
+
+class Archive(object):
```

Comment:
Do you think keeping it separate helps isolating code with specific responsibilities?
From my POV keeping it separate helps minimize the changes on existing code, and open the possibility to enhance in the futurte, for example by using an sqlite db instead of plaintext file.

Reply:
There is really no need in that at the moment. The need will arise when this download archive v2 based on standalone db or whatever is about to be implemented. For now I'd prefer changing couple lines in existing code over premature refactoring even though it may not meet best OOP practices.

Reply:
Ok clear, I have moved logic back to YoutubeDL class